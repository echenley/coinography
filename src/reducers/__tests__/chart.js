import chartReducer from '../chart'

const data = [
  [1520110800, 11259.7, 11293.58, 11273.18, 11260.79, 20.929171180000008],
  [1520110500, 11248.99, 11300.43, 11248.99, 11273.18, 29.644797649999997],
  [1520110200, 11233.58, 11251, 11237.93, 11248.99, 23.785362620000004],
  [1520109900, 11218.71, 11299.31, 11299.3, 11237.99, 43.71891207999998],
  [1520109600, 11271.28, 11315, 11290, 11299.3, 26.03891608000001],
]

const initialState = chartReducer(undefined, {})

describe('chart reducer', () => {
  it('returns correct initialState', () => {
    expect(initialState).toMatchSnapshot()
  })

  it('updates loadingState and requestId on FETCH_HISTORIC_DATA_LOADING', () => {
    expect(
      chartReducer(initialState, {
        type: 'FETCH_HISTORIC_DATA_LOADING',
        requestId: 1,
      })
    ).toMatchSnapshot()
  })

  it('updates only requestId on FETCH_HISTORIC_DATA_LOADING when action.silent === true', () => {
    expect(
      chartReducer(
        {
          ...initialState,
          data,
          loadingState: 'success',
        },
        {
          type: 'FETCH_HISTORIC_DATA_LOADING',
          requestId: 2,
          silent: true,
        }
      ).loadingState
    ).toBe('success')
  })

  describe('FETCH_HISTORIC_DATA_FAILURE', () => {
    it('skips action if requestId does not match', () => {
      const state = chartReducer(initialState, {
        type: 'FETCH_HISTORIC_DATA_LOADING',
        requestId: 2,
        silent: false,
      })

      expect(
        chartReducer(state, {
          type: 'FETCH_HISTORIC_DATA_FAILURE',
          error: {},
          requestId: 1,
        })
      ).toBe(state)
    })

    it('updates loadingState if requestId matches', () => {
      const state = chartReducer(initialState, {
        type: 'FETCH_HISTORIC_DATA_LOADING',
        requestId: 2,
        silent: false,
      })

      expect(
        chartReducer(state, {
          type: 'FETCH_HISTORIC_DATA_FAILURE',
          error: {},
          requestId: 2,
        })
      ).toMatchSnapshot()
    })
  })

  describe('FETCH_HISTORIC_DATA_SUCCESS', () => {
    it('skips action if requestId does not match', () => {
      const state = chartReducer(initialState, {
        type: 'FETCH_HISTORIC_DATA_LOADING',
        requestId: 2,
        silent: false,
      })

      expect(
        chartReducer(state, {
          type: 'FETCH_HISTORIC_DATA_SUCCESS',
          data,
          requestId: 1,
        })
      ).toBe(state)
    })

    it('updates data and loadingState if requestId matches', () => {
      const state = chartReducer(initialState, {
        type: 'FETCH_HISTORIC_DATA_LOADING',
        requestId: 2,
        silent: false,
      })

      expect(
        chartReducer(state, {
          type: 'FETCH_HISTORIC_DATA_SUCCESS',
          data,
          requestId: 2,
        })
      ).toMatchSnapshot()
    })
  })

  it('updates chart type on UPDATE_CHART_TYPE', () => {
    expect(
      chartReducer(initialState, {
        type: 'UPDATE_CHART_TYPE',
        chartType: 'line',
      })
    ).toMatchSnapshot()
  })

  it('updates currency on UPDATE_CURRENCY', () => {
    expect(
      chartReducer(initialState, {
        type: 'UPDATE_CURRENCY',
        currency: 'ETH-USD',
      })
    ).toMatchSnapshot()
  })

  it('updates granularity on UPDATE_GRANULARITY', () => {
    expect(
      chartReducer(initialState, {
        type: 'UPDATE_GRANULARITY',
        granularity: 3600,
      })
    ).toMatchSnapshot()
  })
})
