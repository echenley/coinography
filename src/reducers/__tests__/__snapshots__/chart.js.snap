// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chart reducer FETCH_HISTORIC_DATA_FAILURE updates loadingState if requestId matches 1`] = `
Object {
  "currency": "BTC-USD",
  "data": null,
  "granularity": 300,
  "loadingState": "failure",
  "requestId": 2,
  "type": "candle",
}
`;

exports[`chart reducer FETCH_HISTORIC_DATA_SUCCESS updates data and loadingState if requestId matches 1`] = `
Object {
  "currency": "BTC-USD",
  "data": Array [
    Array [
      1520110800,
      11259.7,
      11293.58,
      11273.18,
      11260.79,
      20.929171180000008,
    ],
    Array [
      1520110500,
      11248.99,
      11300.43,
      11248.99,
      11273.18,
      29.644797649999997,
    ],
    Array [
      1520110200,
      11233.58,
      11251,
      11237.93,
      11248.99,
      23.785362620000004,
    ],
    Array [
      1520109900,
      11218.71,
      11299.31,
      11299.3,
      11237.99,
      43.71891207999998,
    ],
    Array [
      1520109600,
      11271.28,
      11315,
      11290,
      11299.3,
      26.03891608000001,
    ],
  ],
  "granularity": 300,
  "loadingState": "success",
  "requestId": 2,
  "type": "candle",
}
`;

exports[`chart reducer returns correct initialState 1`] = `
Object {
  "currency": "BTC-USD",
  "data": null,
  "granularity": 300,
  "loadingState": "loading",
  "requestId": null,
  "type": "candle",
}
`;

exports[`chart reducer updates chart type on UPDATE_CHART_TYPE 1`] = `
Object {
  "currency": "BTC-USD",
  "data": null,
  "granularity": 300,
  "loadingState": "loading",
  "requestId": null,
  "type": "line",
}
`;

exports[`chart reducer updates currency on UPDATE_CURRENCY 1`] = `
Object {
  "currency": "ETH-USD",
  "data": null,
  "granularity": 300,
  "loadingState": "loading",
  "requestId": null,
  "type": "candle",
}
`;

exports[`chart reducer updates granularity on UPDATE_GRANULARITY 1`] = `
Object {
  "currency": "BTC-USD",
  "data": null,
  "granularity": 3600,
  "loadingState": "loading",
  "requestId": null,
  "type": "candle",
}
`;

exports[`chart reducer updates loadingState and requestId on FETCH_HISTORIC_DATA_LOADING 1`] = `
Object {
  "currency": "BTC-USD",
  "data": null,
  "granularity": 300,
  "loadingState": "loading",
  "requestId": 1,
  "type": "candle",
}
`;
